lib/hypervisor/hv_kvm/__init__.py

```diff
@@ -2554,12 +2555,15 @@ def MigrateInstance(self, cluster_name, instance, target, live_migration):
     if not live_migration:
       self.qmp.StopGuestEmulation()
 
-    max_bandwidth_in_bytes = instance.hvparams[constants.HV_MIGRATION_BANDWIDTH] * 1024 * 1024
-    self.qmp.SetMigrationParameters(max_bandwidth_in_bytes, instance.hvparams[constants.HV_MIGRATION_DOWNTIME])
+    max_bandwidth_in_bytes = \
+        instance.hvparams[constants.HV_MIGRATION_BANDWIDTH] * 1024 * 1024
+    self.qmp.SetMigrationParameters(max_bandwidth_in_bytes,
+        instance.hvparams[constants.HV_MIGRATION_DOWNTIME])
 
     migration_caps = instance.hvparams[constants.HV_KVM_MIGRATION_CAPS]
+    migration_caps = migration_caps.split(_MIGRATION_CAPS_DELIM)
     if migration_caps:
-      self.qmp.SetMigrationCapabilities(migration_caps.split(_MIGRATION_CAPS_DELIM))
+      self.qmp.SetMigrationCapabilities(migration_caps)
 
```

Comment:
could the broken live migration be caused by this change? the `split()` was pushed up before the conditional `if migration_caps` . Somehow migration_caps seems to be `''` (according to the error output):
```
ganeti.errors.HypervisorError: kvm: error executing the migrate-set-capabilities command: Invalid parameter '' (GenericError):
```

Reply:
Well spotted, I'll fix that!

Reply:
I can confirm that #1472 has fixed the issue, well done :-)